doc/HOST_PHASES.md

```diff
@@ -0,0 +1,99 @@
+
+If the sentinel file is not found, the `elemental-agent` will invoke the `OSPlugin` apply the bootstrap config on the system.  
+If the application is successful, the `elemental-agent` will reboot the system to execute the bootstrap config at boot stage.  
+
+Note that the `OSPlugin` can also return an error during this step if it determines that bootstrap was already applied, for example after the reboot.  
+This will lead to the `ElementalHost` being stuck in the `Bootstrapping` phase with an error status condition, requiring human intervention to determine and solve the cause of bootstrap failure.  
+
+The bootstrap configuration is dependent on the [CAPI Bootstrap Provider](https://cluster-api.sigs.k8s.io/reference/providers#bootstrap) in use.  
+
+### Running
+
+The `Running` phase defines the normal operation of the `ElementalHost`.  
+An `ElementalHost` not associated to any `ElementalMachine` and not bootstrapped yet, may simply be `Running` idle, waiting for association.  
+
+If association already happened, then the `ElementalHost` is `Running` without issues and performing its normal operations.  
+
+### Trigger Reset
+
+The `Trigger Reset` phase happens in the following cases:
+
+- The `ElementalHost` is associated to an `ElementalMachine` and the `ElementalMachine` is deleted (for ex. during [machine rollout](https://cluster-api.sigs.k8s.io/tasks/upgrading-clusters#how-to-schedule-a-machine-rollout))  
+- The `ElementalHost` is associated to an `ElementalMachine` belonging to a CAPI `Cluster` and the entire `Cluster` is deleted.  This will lead to the `ElementalMachine` deletion.  
+- The `ElementalHost` is directly deleted. If the `ElementalHost` was associated to an `ElementalMachine`, a new available `ElementalHost` will be picked as replacement.  
+
+During this phase, the `elemental-agent` will inform the `OSPlugin` that reset has been triggered.  
+Implementation details are plugin dependent, this is the occasion for the plugin to stop services and do anything needed to prepare the system for a reset.  
+
+The `elemental-agent` will exit once reset has been triggered.  
+It is expected to run `elemental-agent --reset` after, to actually perform the reset of the host.  
+
+### Resetting
+
+The `Resetting` phase happens when `elemental-agent --reset` is ran.  
+
+The `elemental-agent` will first delete the remote `ElementalHost`.  
+This will add a `deletionTimestamp` to the remote resource, however a finalizer will prevent deletion until reset is deemed successful.  
+
+After deleting the remote `ElementalHost`, the `elemental-agent` will fetch the remote `ElementalRegistration` and pass the schemaless `ElementalRegistration.spec.elemental.reset` field to the `OSPlugin` to perform reset.  
+
+If the `OSPlugin` reset the host successfully, the remote `ElementalHost` is updated one last time to highlight reset has been completed. This will allow the deletion of the `ElementalHost`.  
```

Comment:
```suggestion
If the `OSPlugin` resets the host successfully, the remote `ElementalHost` is updated one last time to highlight reset has been completed. This will allow the deletion of the `ElementalHost`.  
```